src/v/cluster/scheduling/allocation_node.h

```diff
@@ -89,20 +93,22 @@ class allocation_node {
     }
     bool is_full() const { return _allocated_partitions >= _max_capacity; }
     allocation_capacity max_capacity() const { return _max_capacity; }
-    ss::shard_id allocate();
+    ss::shard_id allocate(bool prio);
```

Comment:
This function needs a clear comment about the meaning of priority. Chose one place to put a clear description of what "priority" means and then the per-function comments can be a bit more terse at least, but we should have one.

Reply:
I'm going convert the `bool` flag into an enum. I think that will be the right place for documenting that.